k8s/core/core.go

```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/portworx/sched-ops/k8s/common"
 	"github.com/portworx/sched-ops/task"
 	"github.com/sirupsen/logrus"
+	certv1 "k8s.io/api/certificates/v1"
```

Comment:
We have been creating separate clients under k8s folder for different API groups. Core, rbac, policy, etc.
Let's do the same for certificate APIs for consistency?

Reply:
> We have been creating separate clients under k8s folder for different API groups. Core, rbac, policy, etc. Let's do the same for certificate APIs for consistency?

@piyush-nimbalkar -- not sure what you are asking me to do here.  Can you give me an example?

Note that I had do add small changes into `core.go`, to include the CSR interface into the main sched-ops client, and also to handle the CSR watches.  The implementation went into a separate `csr.go` file.

Reply:
I meant move it to a separate package just like core and not just to a separate file - `k8s/certificates/csr.go`
We did that for all the different k8s apis.

Reply:
> We did that for all the different k8s apis

I checked the [core.handleWatch()](https://github.com/portworx/sched-ops/blob/master/k8s/core/core.go#L225) -- and `WatchNode()`, `WatchConfigMap()`, `WatchPods()` and `WatchSecret()` are all implemented in the `core` package.
* since `Certificates` is IMO a concept very close to `Secrets` -- I followed how/where the `Secrets` API was implemented, and thus added it into the `core` package
* on the other hand, k8s-subdirs seem to be product-related  (`k8s/{stork,talisman,openshift,prometheus,autopilot}`...)

Is this going to be a problem?

Reply:
> k8s-subdirs seem to be product-related

That's not true. We have packages like rbac, apps, storage, batch, policy, dynamic, etc. All of them are separated by the Kubernetes API. Just because the concept is close to Secrets does not mean we should make an exception.

Of course, it is not going to a problem because this won't vendor new outside packages (one of the reasons why we separated these packages). It is just about consistency.

I have approved the PR already, so will leave it up to you. Just wanted to point out the inconsistency.